src/App.tsx

```diff
@@ -1,26 +1,48 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from './app/hooks';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { setTodos } from './features/todos';
+import { selectCurrentTodo, selectTodos } from './features/todoSelectors';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App: React.FC = () => {
+  const dispatch = useDispatch();
+  const todos = useAppSelector(selectTodos);
+  const currentTodo = useAppSelector(selectCurrentTodo);
+  const [isLoading, setIsLoading] = useState(false);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(getTodo => {
+        dispatch(setTodos(getTodo));
+      })
+      .finally(() => setIsLoading(false));
```

Comment:
It's better to move the logic inside the useEffect hook to a separate function, as it improves code readability, reusability, and makes testing easier. It also helps keep useEffect focused on side effects.